cfme/tests/infrastructure/test_datastore_analysis.py

```diff
@@ -30,130 +27,96 @@
 
 
 def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = testgen.providers_by_class(
+        metafunc, [RHEVMProvider, VMwareProvider], required_fields=['datastores'])
+    argnames.append('datastore_type')
     new_idlist = []
     new_argvalues = []
 
-    argnames, argvalues, idlist = testgen.providers_by_class(
-        metafunc, PROVIDER_TYPES, required_fields=['datastores'])
-    argnames += ['datastore']
-    appliance = get_or_create_current_appliance()
-    # TODO: turn the datastore into a parameterized fixture by type,
-    #       and discuss semantics for obtaining them by type
-    datastore_collection = datastore.DatastoreCollection(appliance)
-
-    for i, argvalue_tuple in enumerate(argvalues):
+    for index, argvalue_tuple in enumerate(argvalues):
         args = dict(zip(argnames, argvalue_tuple))
-        datastores = args['provider'].data.get('datastores', {})
-        if not datastores:
-            continue
+        datastores = args['provider'].data.datastores
         for ds in datastores:
-            if not ds.get('test_fleece', False):
+            if not ds.test_fleece:
```

Comment:
**Question/Required**

Why drop the `.get` call here, opening this to an AttributeError exception?

Reply:
done, restored .get  call